enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,405 @@
+
+Essentially, when operator authors are declaring support for multiple architectures, that entails:
+
+- Making a declaration of that support via the CSV architecture annotation
+- Setting node affinity in the CSV deployment spec so that the operator is scheduled correctly.
+- Update the operator code such that when it creates/updates operand workloads, it sets those same nodeAffinity settings that are set on
+  the operator deployment.
+
+### Machine Config Operator
+
+No changes are needed in the MCO for Phase 1 as the machine-os-content image would be a manifest listed image and the machine config daemon would
+extract the relevant architecture's machine-os-content based on the node it runs on. In the future, architecture specific configurations should
+not be required, instead they need to be templatized and be generalized for certain items, for example kubelet system reserved memory could scale
+based on page size.
+
+### Machine API
+
+- As of today, machinesets can be mutated to change the instance type and image id and this could potentially result in inconsistency between existing
+machines and newly created machines. If the instance type and image is changed to one of a different architecture, there would be workers of
+different architectures with the same machineset definition. While this may not be a problem, it doesn't make for a good user experience. Hence,
+machineset specs will not allowed to be mutated. Anytime a change is required, a new machineset needs to be created and the old one needs to be scaled
+down.
+- Changes are needed in the autoscaler to make it aware of the node architecture when autoscaling from zero. The `kubernetes-io-arch` label needs to be
+set when building the NodeInfo object so the autoscaler can pick the correct pool to scale up for allocating nodes of a particular architecture.
+
+### Hypershift
+
+The addition of heterogeneous support in Openshift also opens up a new dimension for [support][OCPPLAN-5684] in Hypershift where the control plane 
+and the workers could run on different architectures. There would be certain changes needed to realize this as there are limitations which need to
+be addressed:
+
+- When creating the hosted cluster, the latest x86 release payload is picked by default. This would have to change to pick a heterogeneous payload
+when creating the hosted cluster on mixed architecture workers.
+- For the hosted control plane to work on non x86 architecture worker nodes, the following image references would need changing:
+  - the release image fixtures referencing x86 imagestreams. these would need to be changed to manifestlist references or there would need
+  to be imagestreams added for individual architectures and they would be picked based on the node architecture.
+  - references to some images like cluster-api-provider-aws, apiserver-network-proxy etc.. are x86. these would need to change to provide the images
```

Comment:
apiserver-network-proxy comes from the payload which is dictated by hostedCluster.spec.release.image, so as far as that's heterogeneous this image should be fine.

Reply:
yes i'll change this. at the time of writing , it didn't come from the payload but since then this PR: https://github.com/openshift/hypershift/pull/1215 take care of it